processor/src/main/resources/org/mapstruct/ap/internal/model/IterableMappingMethod.ftl

```diff
@@ -54,7 +54,8 @@
 
     <#if resultType.arrayType>
         <#if !existingInstanceMapping>
-            <@includeModel object=resultElementType/>[] ${resultName} = new <@includeModel object=resultElementType/>[<@iterableSize/>];
+            <#assign elementTypeString><@includeModel object=resultElementType/></#assign>
```

Comment:
should we do this kind of manipulation in the FreeMarker template or in the java code?